torch/testing/_internal/common_device_type.py

```diff
@@ -1167,6 +1168,32 @@ def skipCUDAIfRocm(fn):
 def skipCUDAIfNotRocm(fn):
     return skipCUDAIf(not TEST_WITH_ROCM, "test doesn't currently work on the CUDA stack")(fn)
 
+# Skips a test on CUDA if ROCm is unavailable or its version is lower than requested.
+def skipCUDAIfRocmVersionLessThan(version=None):
+
+    def dec_fn(fn):
+        @wraps(fn)
+        def wrap_fn(self, *args, **kwargs):
+            if self.device_type == 'cuda':
+                if not TEST_WITH_ROCM:
+                    reason = "ROCm not available"
+                    raise unittest.SkipTest(reason)
```

Comment:
I think this will cause CUDA to skip any test using this decorator because CUDA never sets TEST_WITH_ROCM.  I think the logic you were going for was

```
if self.device_type == 'cuda' and torch.version.hip is not None
    # rocm version parsing etc
```

Reply:
I am not sure how the ROCm CI sets `PYTORCH_TEST_WITH_ROCM=1`, but `TEST_WITH_ROCM` is just the internal representation of that flag (read in `common_utils.py`). I got the idea to use this flag from the `skipCUDAIfNotRocm()` function just a few lines above. 

When I am testing, I am manually setting `PYTORCH_TEST_WITH_ROCM`, as in the description of this PR.